zsh-syntax-highlighting.zsh

```diff
@@ -27,571 +27,9 @@
-# Core highlighting update system
-# -------------------------------------------------------------------------------------------------
-
-# Use workaround for bug in ZSH?
-# zsh-users/zsh@48cadf4 http://www.zsh.org/mla/workers//2017/msg00034.html
-autoload -Uz is-at-least
-if is-at-least 5.4; then
-  typeset -g zsh_highlight__pat_static_bug=false
-else
-  typeset -g zsh_highlight__pat_static_bug=true
-fi
-
-# Array declaring active highlighters names.
-typeset -ga ZSH_HIGHLIGHT_HIGHLIGHTERS
-
-# Update ZLE buffer syntax highlighting.
-#
-# Invokes each highlighter that needs updating.
-# This function is supposed to be called whenever the ZLE state changes.
-_zsh_highlight()
-{
-  # Store the previous command return code to restore it whatever happens.
-  local ret=$?
-  # Make it read-only.  Can't combine this with the previous line when POSIX_BUILTINS may be set.
-  typeset -r ret
-
-  # $region_highlight should be predefined, either by zle or by the test suite's mock (non-special) array.
-  (( ${+region_highlight} )) || {
-    echo >&2 'zsh-syntax-highlighting: error: $region_highlight is not defined'
-    echo >&2 'zsh-syntax-highlighting: (Check whether zsh-syntax-highlighting was installed according to the instructions.)'
-    return $ret
-  }
-
-  # Probe the memo= feature, once.
-  (( ${+zsh_highlight__memo_feature} )) || {
-    region_highlight+=( " 0 0 fg=red, memo=zsh-syntax-highlighting" )
-    case ${region_highlight[-1]} in
-      ("0 0 fg=red")
-        # zsh 5.8 or earlier
-        integer -gr zsh_highlight__memo_feature=0
-        ;;
-      ("0 0 fg=red memo=zsh-syntax-highlighting")
-        # zsh 5.9 or later
-        integer -gr zsh_highlight__memo_feature=1
-        ;;
-      (" 0 0 fg=red, memo=zsh-syntax-highlighting") ;&
-      (*)
-        # We can get here in two ways:
-        #
-        # 1. When not running as a widget.  In that case, $region_highlight is
-        # not a special variable (= one with custom getter/setter functions
-        # written in C) but an ordinary one, so the third case pattern matches
-        # and we fall through to this block.  (The test suite uses this codepath.)
-        #
-        # 2. When running under a future version of zsh that will have changed
-        # the serialization of $region_highlight elements from their underlying
-        # C structs, so that none of the previous case patterns will match.
-        #
-        # In either case, fall back to a version check.
-        #
-        # The memo= feature was added to zsh in commit zsh-5.8-172-gdd6e702ee.
-        # The version number at the time was 5.8.0.2-dev (see Config/version.mk).
-        # Therefore, on 5.8.0.3 and newer the memo= feature is available.
-        #
-        # On zsh version 5.8.0.2 between the aforementioned commit and the
-        # first Config/version.mk bump after it (which, at the time of writing,
-        # is yet to come), this condition will false negative.
-        if is-at-least 5.8.0.3; then
```

Comment:
I just pushed 3944a44ffe44 which changes two lines, including this one.  Sorry to have caused you conflicts, but it was a hotfix needed to unbreak z-asug under 5.8; see #756.

Reply:
No worries. This will create conflicts for everything else and bug fixes come before features. Rebased.